src/examples/code/perl.man
.P
.RS 2
.nf
# loads object
sub load
{
  my $flds = $c->db_load($id,@_) || do {
    Carp::carp "Can`t load (class: $c, id: $id): '$!'"; return undef
  };
  my $o = $c->_perl_new();
  $id12 = $id / 24 / 3600;
  $o->{'ID'} = $id12 + 123;
  #$o->{'SHCUT'} = $flds->{'SHCUT'};
  my $p = $o->props;
  my $vt;
  $string =~ m/^sought_text$/;
  $items = split //, 'abc';
  $string //= "bar";
  for my $key (keys %$p)
  {
    if(${$vt.'::property'}) {
      $o->{$key . '_real'} = $flds->{$key};
      tie $o->{$key}, 'CMSBuilder::Property', $o, $key;
    }
  }
  $o->save if delete $o->{'_save_after_load'};

  # GH-117
  my $g = glob("/usr/bin/*");

  return $o;
}

__DATA__
@@ layouts/default.html.ep
<!DOCTYPE html>
<html>
  <head><title><%= title %></title></head>
  <body><%= content %></body>
</html>
__END__

=head1 NAME
POD till the end of file
.fi
.RE

Changelog.md
Version changes
=================================================

The following list gives a short overview about what is changed between
individual versions:

Version 2.2.2 (2016-12-10)
-------------------------------------------------
- Fixed test for title in table of contents.
- Remove empty lines from handlebars templates.
- Update phantomjs-prebuilt@2.1.14
- Fixed: Tables were being removed in the process of putting boxes into tabs.
- Optimize table of contents display and box titles.
- Smaller fixes in html style.

Version 2.2.1 (2016-12-08)
-------------------------------------------------
- Bug fix include of trans submodule.
- Add documentation of themes.
- Fix links of general examples.

Version 2.2.0 (2016-12-08)
-------------------------------------------------
Boxes will stack together as tabbed box.

- Allow longer runtime for plantuml.
- Remove support for nodejs 0.10 because submodules no longer support it.
- Fix tests to work with variing html.
- Add top link in table of contents.
- Allow all tests to run again.
- Add implementation of output specific parts.
- Allow initialization with markdown.
- Remove unneccessary parts from documentation.
- Set default height for boxes to 300px.
- Added examples for tables within boxes.
- Finished code in boxes examples.
- Fix console output for boxed code.
- Colorize code tabs.
- Change html for code in boxes.
- Added tests for all box possibilities.
- Update async@2.1.4, clean-css@3.4.21, handlebars@4.0.6, alinex-util@2.5.0, debug@2.3.3, highlight.js@9.8.0, markdown-it@8.2.1, deasync@0.1.9
- Meke test for simple boxes.
- Inline document some more functions.
- Optimize console output of boxes.
- Extract translations into extra file.
- Finish box layout with tabs for html.
- Add color style to default boxes.
- Add print layout for boxes.
- Add simple tab css and upgrade fontawesome to 4.7.0
- Update npm markdown-it@8.2.1
- Change box transformation to create propper html for tabbed boxes.
- Transform box to tabs html structure.
- Merge branch 'master' of https://github.com/alinex/node-report
- Description of new code and box elements.
- Add testlayout for new box model.
- Add info for documentation which is created using the module itself.
- Fix blockquote nesting.
- Update style for boxes and more.
- Add example for stacked quotes.
- Optimize style of blockquotes.
- Create new documentation examples.
- Updated documentation.
- Remove unused style.
- Add header to detail box and optimize console output of boxes.
- Update alinex-fs@3.0.3 markdown-it@8.1.0 markdown-it-attrs@0.8.0 node-plantuml@0.5.0
- Create debug messages only if enabled.

Version 2.1.12 (2016-10-19)
-------------------------------------------------
- Remove line breaks before html comments for rendering.
- Describe github comment handling.
- Updated ignore files.
- Update travis.
- Update travis.

Version 2.1.11 (2016-08-18)
-------------------------------------------------
- Update html-pdf@2.1.0
- Fix test to allow slightly change of footnotes.
- Update markdown-it-attrs@0.6.3
- Update alinex-fs@2.0.7, markdown-it@7.0.1, markdown-it-footnote@3.0.1, phantomjs-prebuilt@2.1.12, alinex-builder@2.3.6, alinex-util@2.4.0, highlight.js@9.6.0
- Downgrade markdown-it to 6.1.0 because of markdown-it-attr dependency.
- Add support for alternative box titles.

Version 2.1.10 (2016-08-02)
-------------------------------------------------
- Upgraded to phantomjs-prebuilt@2.1.9.
- Allow direct styles through api.
- Allow direct styles in each elements using curly braces.

Version 2.1.9 (2016-07-25)
-------------------------------------------------
- Support includes in handlebar templates.
- Allow handlebar template includes.

Version 2.1.8 (2016-07-21)
-------------------------------------------------
- Correct line number positioning to block before not at start.

Version 2.1.7 (2016-07-21)
-------------------------------------------------
- Upgrade alinex-fs@2.0.6, alinex-builder@2.3.1
- fix decorator problem for code elements

Version 2.1.6 (2016-07-19)
-------------------------------------------------
- Remove unneccessary modules.
- Add header to table of contents.

Version 2.1.5 (2016-07-19)
-------------------------------------------------
- Fix test for new code block structure.
- Update all language formats.
- Keep multiline code highliting intact.
- Add support for stylus and css.
- Move local links to codedoc.
- Better javascript and local link support.

Version 2.1.4 (2016-07-15)
-------------------------------------------------
- Fixed style links.

Version 2.1.3 (2016-07-15)
-------------------------------------------------
- Fixed style links.

Version 2.1.2 (2016-07-15)
-------------------------------------------------
- Updated test to work with new html structure.
- Allow style on code tag be transported to pre tag before.
- Rename links to Alinex Namespace.
- Move codedoc template into it's own package.
- Fix extraction of title with starting alternative heading.
- Optimize Style.
- Update style for codedoc sidebar.
- Upgraded alinex-handlebars@1.2.1, alinex-builder@2.2.1, async@2.0.0
- Show file tree from context in handlebars template for codedoc.
- Add getReport method.
- Add files from context.
- Allow additional context for handlebars.

Version 2.1.1 (2016-07-12)
-------------------------------------------------
- Add inline styles for codedoc schema.
- Finish responsive top buttons.
- Create bootstrap layout for codedoc.
- Make own codedoc layout.

Version 2.1.0 (2016-07-11)
-------------------------------------------------
- Upgradet alinex-fs@2.0.3, alinex-util@2.3.1, alinex-builder@2.2.0
- Support handlebars for styles.
- Make handlebars template working.
- Use template search for css and handlebars.
- Add addintional style.
- Small code updates.

Version 2.0.0 (2016-06-30)
-------------------------------------------------
Now with support for multiple graphical visualization, restructered documentation and more.


src/examples/code/perl.tokens.js
TokenList {
  data: 
   [ { type: 'document', nesting: 1, level: 0 },
     { type: 'code',
       nesting: 1,
       language: 'perl',
       level: 1,
       parent: [Object] },
     { type: 'text',
       content: '# loads object\nsub load\n{\n  my $flds = $c->db_load($id,@_) || do {\n    Carp::carp "Can`t load (class: $c, id: $id): \'$!\'"; return undef\n  };\n  my $o = $c->_perl_new();\n  $id12 = $id / 24 / 3600;\n  $o->{\'ID\'} = $id12 + 123;\n  #$o->{\'SHCUT\'} = $flds->{\'SHCUT\'};\n  my $p = $o->props;\n  my $vt;\n  $string =~ m/^sought_text$/;\n  $items = split //, \'abc\';\n  $string //= "bar";\n  for my $key (keys %$p)\n  {\n    if(${$vt.\'::property\'}) {\n      $o->{$key . \'_real\'} = $flds->{$key};\n      tie $o->{$key}, \'CMSBuilder::Property\', $o, $key;\n    }\n  }\n  $o->save if delete $o->{\'_save_after_load\'};\n\n  # GH-117\n  my $g = glob("/usr/bin/*");\n\n  return $o;\n}\n\n__DATA__\n@@ layouts/default.html.ep\n<!DOCTYPE html>\n<html>\n  <head><title><%= title %></title></head>\n  <body><%= content %></body>\n</html>\n__END__\n\n=head1 NAME\nPOD till the end of file',
       level: 2,
       parent: [Object] },
     { type: 'code',
       nesting: -1,
       language: 'perl',
       level: 1,
       parent: [Object] },
     { type: 'document', nesting: -1, level: 0 } ],
  pos: 4,
  token: 
   { type: 'code',
     nesting: -1,
     language: 'perl',
     level: 1,
     parent: { type: 'document', nesting: 1, level: 0 } } }